ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -289,6 +289,7 @@ private boolean validateTxnList() throws CommandProcessorException {
           }
           // Since we're reusing the compiled plan, we need to update its start time for current run
           driverContext.getPlan().setQueryStartTime(driverContext.getQueryDisplay().getQueryStartTime());
+          driverContext.setRetrial(false);
```

Comment:
why do we need this?

Reply:
The retrial flag is needed to re-create the plan, but when the plan is re-created the flag needs to be cleared, otherwise the next query will fail. This is a bug I found by the testRebalanceCompactionWithParallelDelete() test.

Reply:
👍